CDCgov/prime-reportstream
# Initial integration of fhir transforms into universal pipeline

This PR adds code to FHIRConverter.kt to perform sender transforms. Part of enabling this involved adding a schema name both to the sender configuration and the `RawSubmission` object that gets passed to the message queue. Other than that, things here worked pretty seamlessly.

Test Steps:
1. Pull the branch and add ```      schemaName: metadata/fhir_transforms/fhir-transform-sample``` to a full-elr sender. Then reload settings
2. Run ReportStream locally and send a report to that sender.
3. Either on the receiver side, or in blob storage, look for some representation of the report from after the convert step. Search for `TestValue` to see that the patient name was converted to the value from the transform.

## Changes
- Added `schemaName` to `ELRSender` to track a Fhir transform schema name in sender settings
- In the convert step, before looping through fhir bundles, load the Fhir transformer, and transform the bundles
- Removed `HasSchema` interface (didn't seem to provide any value since it just had a simple property), was formerly used to identify `TopicSender` so updated some of those checks.
- Added tests that verify that `transform` is called during the convert step.

## Checklist

### Testing

## Linked Issues
- Addresses #7252
